src/hooks/useOnboard.ts

```diff
@@ -0,0 +1,125 @@
+import React from "react";
+import { Wallet } from "bnc-onboard/dist/src/interfaces";
+import Onboard from "bnc-onboard";
+import { Initialization } from "bnc-onboard/dist/src/interfaces";
+
+import { useConnection } from "./useConnection";
+
+const infuraId = process.env.REACT_APP_PUBLIC_INFURA_ID;
+
+const getNetworkName = (chainId: number) => {
+  switch (chainId) {
+    case 1: {
+      return "mainnet";
+    }
+    case 42: {
+      return "kovan";
+    }
+    case 3: {
+      return "ropsten";
+    }
+    case 4: {
+      return "rinkeby";
+    }
+  }
+};
+
+function onboardBaseConfig(_chainId?: number): Initialization {
+  const chainId = _chainId ?? 1;
+  const infuraRpc = `https://${getNetworkName(
+    chainId
+  )}.infura.io/v3/${infuraId}`;
+
+  return {
+    dappId: process.env.REACT_APP_ONBOARD_API_KEY || "",
+    hideBranding: true,
+    networkId: 1, // Default to main net. If on a different network will change with the subscription.
+    walletSelect: {
+      wallets: [
+        { walletName: "metamask", preferred: true },
+        {
+          walletName: "imToken",
+          rpcUrl:
+            chainId === 1
+              ? "https://mainnet-eth.token.im"
+              : "https://eth-testnet.tokenlon.im",
+          preferred: true,
+        },
+        { walletName: "coinbase", preferred: true },
+        {
+          walletName: "portis",
+          apiKey: process.env.NEXT_PUBLIC_PORTIS_API_KEY,
+        },
+        { walletName: "trust", rpcUrl: infuraRpc },
+        { walletName: "dapper" },
+        {
+          walletName: "walletConnect",
+          rpc: { [chainId || 1]: infuraRpc },
+        },
+        { walletName: "gnosis" },
+        { walletName: "walletLink", rpcUrl: infuraRpc },
+        { walletName: "opera" },
+        { walletName: "operaTouch" },
+        { walletName: "torus" },
+        { walletName: "status" },
+        { walletName: "unilogin" },
+        {
+          walletName: "ledger",
+          rpcUrl: infuraRpc,
+        },
+      ],
+    },
+    walletCheck: [
+      { checkName: "connect" },
+      { checkName: "accounts" },
+      { checkName: "network" },
```

Comment:
not sure how this network check works, but it might interfeerre with these requests on other chains